service/pokemon.go

```diff
@@ -0,0 +1,115 @@
+package service
+
+import (
+	"encoding/csv"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+
+	"github.com/eugenio-garcia/Go-Dispatch-Bootcamp/errors"
+	"github.com/eugenio-garcia/Go-Dispatch-Bootcamp/model"
+)
+
+type PokemonMap map[int]model.Pokemon
+
+var pokemonsOrder []int = []int{1, 2, 3}
+
+var db PokemonMap = map[int]model.Pokemon{
+	1: {ID: 1, Name: "Pikachu"},
+	2: {ID: 2, Name: "Bulbasur"},
+}
+
+type PokemonService struct {
+	data PokemonMap
+}
+
+func New(pm PokemonMap) *PokemonService {
+	if pm == nil {
+		pm = OpenCSV()
+	}
+
+	return &PokemonService{
+		data: pm,
+	}
+}
+
+type PokemonRecord struct {
+	ID   int
+	Name string
+}
+
+func createPokemonMap(data [][]string) map[int]model.Pokemon {
+	var pokemonList PokemonMap = map[int]model.Pokemon{}
+	for i, line := range data {
+		if i > 0 { // omit header line
+			var rec PokemonRecord
+			for j, field := range line {
+				if j == 0 {
+					rec.ID, _ = strconv.Atoi(field) //Error?
+				} else if j == 1 {
+					rec.Name = field
+				}
+			}
+			pokemon := model.Pokemon{
+				ID:   rec.ID,
+				Name: rec.Name,
+			}
+			pokemonList[i] = pokemon
+		}
+	}
+	return pokemonList
+}
+
+func OpenCSV() map[int]model.Pokemon {
```

Comment:
Does this function need to be exported?

Reply:
not , is actually only needed within the service